### DISCUSSION TITLE
Cursor dissapears/dimming when tablet is used for clicking

### DISCUSSION THREAD

**Discussion by RBRat3** [ppy/osu#21202] [Q&A] [answered]
### Type

Cosmetic

### Bug description

Dont know if this is a bug or a feature but if it's the later I dont know who it's for. The cursor when a tablet pen press is used to click a hit object the cursor alpha fades, on long runs or sliders there is barely a cursor visible. This some what hurts players who do pen tapping or those who are new and as for me is unplayable at high difficulties as I've always pen tapped. There doesn't seem to be a setting to disable this to my knowledge.

### Screenshots or videos

https://puu.sh/Jri6e/0afa58b76c.mp4

### Version

2022.1101.0-lazer

### Logs

none

**Comment by peppy** [comment, marked as answer]
This is intentional behaviour for touch devices, because you are directly interacting with the screen.

If you want the cursor to display, please toggle the setting for this:

<img width="403" alt="2022-11-11 15 28 57@2x" src="https://user-images.githubusercontent.com/191335/201277545-84932fba-d77f-419c-8c13-753c67a46915.png">

**Reply by RBRat3** [reply]
Thx!!! That's somewhat buried as I was digging through skinning. Im not using touch input, tho most new pen tabs are capable of multi-touch (intuos 5) so I guess it's a detection thing? Completely missed that cause wasn't thinking of touch :D

Edit: Also to add if this detects the way I think it does and not exclusive to me it affects a lot of wacom tablets across a few lines. Most shut off the touch feature because of the shitty palm detection and forget about it so "touch" wont be the first on their mind for that setting, I probably wont be the only person to ask this question.

**Reply by peppy** [reply]
I see. This suggests that OTD may be misinterpreting the device type. I haven't looked into whether we can better understand the input device type when events arrive, but it could provide us with a way of differentiating actual touch input from non-.

**Comment by LittleEndu** [comment]
Sounds like your tablet might be reporting cursor as touch input. Have you tried turning on the "show cursor on touch input" setting?

**Reply by RBRat3** [reply]
My tablet can do both, along with many newer wacom devices im guessing it's a mixup on device types ¯\_(ツ)_/¯
